developer_docs/provisioning_setup.asciidoc

```diff
@@ -0,0 +1,258 @@
+[[api-v2-development]]
+= Provisioning setup
+:toc: right
+:toclevels: 5
+
+Guide how to setup provisioning on Fedora with Libvirt.
+
+[[tftp-configuration]]
+== TFTP configuration
+
+[source, bash]
+....
+dnf install -y syslinux
+mkdir -p /var/lib/tftpboot
+mkdir -p /var/lib/tftpboot/{boot,grub,grub2,pxelinux.cfg}
+cp /usr/share/syslinux/{pxelinux.0,menu.c32,chain.c32,ldlinux.c32,libcom32.c32,libutil.c32} /var/lib/tftpboot
+
+# Change file owner to your account
+chown your-account:your-group -R /var/lib/tftpboot
+....
+
+[[libvirt-configuration]]
+== Libvirt configuration
+Install Libvirt
+[source, bash]
+....
+dnf install @virtualization
+....
+
+Check that `libvirtd` is running
+[source, bash]
+....
+systemctl start libvirtd
+systemctl enable libvirtd
+systemctl status libvirtd
+....
+
+Edit Libvirt `default` network
+[source, bash]
+....
+# virsh net-edit default
+
+<network>
+  <name>default</name>
+  <uuid>your-uuid</uuid>
+  <forward mode='nat'/>
+  <bridge name='virbr0' stp='on' delay='0'/>
+  <mac address='your-mac'/>
+  <domain name='local.loc' localOnly='yes'/>
+
+  <dns>
+    <host ip='192.168.122.1'>
+      <hostname>foreman.local.loc</hostname>
+    </host>
+  </dns>
+
+  <ip address='192.168.122.1' netmask='255.255.255.0'>
+    <tftp root='/var/lib/tftpboot'/>
+    <dhcp>
+      <range start='192.168.122.2' end='192.168.122.254'/>
+      <bootp file='pxelinux.0'/>
+    </dhcp>
+  </ip>
+</network>
+....
+
+Reload network configuration
+[source, bash]
+....
+virsh net-destroy default
+virsh net-start default
+....
+
+[[network-configuration]]
+== Network configuration
+Configure `dnsmasq` for Libvirt
+[source, bash]
+....
+/etc/NetworkManager/dnsmasq.d/foreman.conf
+
+# All local (example.tst) queries are redirected to libvirt DNS provider
+server=/local.loc/192.168.122.1
+mx-host=local.loc,foreman.local.loc
+ptr-record=1.122.168.192.in-addr.arpa,foreman.local.loc
+
+# This main dnsmasq listens only on local address so libvirt can start its
+# own dnsmasqs for each netwrok
+listen-address=127.0.0.1
+
+# For debugging purposes it's sometimes useful to see queries
+log-queries
+....
+
+In Firewall configure `vibr0` to `trusted` zone
+
+[source, bash]
+....
+firewall-cmd --change-interface virbr0 --zone trusted
+firewall-cmd --change-interface virbr0 --zone trusted --permanent
+nmcli connection modify virbr0 connection.zone trusted
+....
+
+Configure network manager
+[source, bash]
+....
+# /etc/NetworkManager/NetworkManager.conf
+
+[main]
+dns=dnsmasq
+....
+
+Delete the `/etc/resolv.conf` symlink
+[source, bash]
+....
+rm /etc/resolv.conf
+....
+
+Add `foreman.local.loc` to `/etc/hosts`
+
+[source, bash]
+....
+127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4 foreman.local.loc
+::1         localhost localhost.localdomain localhost6 localhost6.localdomain6 foreman.local.loc
+....
+
+Restart computer
+
+[[smart-proxy-configuration]]
+== Smart Proxy configuration
```

Comment:
I really think we should make sure this lives in https://github.com/theforeman/smart-proxy and refer to that documentation. Documenting the Smart Proxy in foreman.git is wrong because it's far removed from the actual code. Instead I think we should encourage people to interact with the smart proxy much more directly and document it in the repository.

That also means that I think that a lot of this setup should live in the Smart Proxy